hal/api/platform.h

```diff
@@ -54,4 +55,52 @@ class PlatformMutex {
 
 #endif
 
+/** The static version of a PlatformMutex
+ *
+ * This class must only be used in a static context -
+ * this class must never be allocated or created on the
+ * stack.
+ *
+ * This class is lazily initialized on first use.
+ * This class is a POD type so if it is not used it will
+ * be garbage collected.
+ */
+struct PlatformMutexStatic {
+    PlatformMutex* _mutex;
+
+    void _init() {
+        PlatformMutex* temp = new PlatformMutex;
+
+        core_util_critical_section_enter();
+        if (NULL == _mutex) {
+            _mutex = temp;
+        }
+        core_util_critical_section_exit();
```

Comment:
Actually looks like cas was already added (by you even :P): https://github.com/mbedmicro/mbed/blame/master/hal/api/critical.h#L210

There's no pointer cas, but the u32 one should work:
``` cpp
PlatformMutex *temp = new PlatformMutex;
PlatformMutex *expected = NULL;

while (!core_util_atomic_cas_u32((uint32_t *)&_mutex, (uint32_t *)&expected, (uint32_t)temp) {
    if (expected != NULL) {
        // Mutex was created on another thread first
        // so delete ours
        delete temp;
    }
}
```